main.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	_ "github.com/lib/pq"
+	"log"
+	"net/http"
+)
+// Database information inserted
+const (
+	DbUser     = "meclpohq"
+	DbPassword = "h6jPxvrJQkTmTYRZge1f0ZSShyEjItzI"
+	DbName     = "meclpohq host=rajje.db.elephantsql.com"
+	DbHost     = "rajje.db.elephantsql.com"
+)
+
+var router *gin.Engine
+var db *sql.DB
+// Structure of database is declared
+type patient struct {
+	ID         int    `json:"id"`
+	Name       string `json:"name"`
+	Illness    string `json:"illness"`
+	BirthDate  string `json:"birth_date"`
+	LocationId int    `json:"location_id"`
+}
+
+type location struct {
+	ID         int    `json:"id"`
+	Name       string `json:"name"`
+	HospitalId int    `json:"hospital_id"`
+}
+
+type hospital struct {
+	ID         int    `json:"id"`
+	Name       string `json:"name"`
+	HospitalId int    `json:"max_patient_amount"`
+}
+// function declaring abouter router
+func main() {
+	router = gin.Default()
+	initDatabase()
+	initAPI()
+	err := router.Run(":8080")
+	if err != nil {
+		log.Fatalln("Could not run the app")
+	}
+	defer db.Close()
+}
+
+func initDatabase() {
+
+	psqlInfo := fmt.Sprintf("user=%s password=%s dbname=%s host=%s", DbUser, DbPassword, DbName, DbHost)
+	var err error
+
+	db, err = sql.Open("postgres", psqlInfo)
+	if err != nil {
+		panic(err)
+	}
+
+	err = db.Ping()
+	if err != nil {
+		panic(err)
+	}
+	log.Println("Successfully connected to the database")
+}
+// Provide details of all patients
+func GetAllPatients(c *gin.Context) {
+
+	rows, err := db.Query("SELECT * FROM patient")
```

Comment:
Use `*sqlx.DB#Select()` method instead, it's more readable.

Reply:
okk